pybind_interface/pybind_main.cpp

```diff
@@ -375,6 +389,59 @@ std::vector<std::complex<float>> qsim_simulate(const py::dict &options) {
   return amplitudes;
 }
 
+std::vector<std::complex<float>> qtrajectory_simulate(const py::dict &options) {
+  NoisyCircuit<Cirq::GateCirq<float>> ncircuit;
+  unsigned num_qubits;
+  std::vector<Bitstring> bitstrings;
+  try {
+    ncircuit = getNoisyCircuit(options);
+    num_qubits = qsim::count_qubits(ncircuit);
+    bitstrings = getBitstrings(options, num_qubits);
+  } catch (const std::invalid_argument &exp) {
+    IO::errorf(exp.what());
+    return {};
+  }
+
+  using Simulator = qsim::Simulator<For>;
+  using StateSpace = Simulator::StateSpace;
+  using State = StateSpace::State;
+
+  // Define container for amplitudes
+  std::vector<std::complex<float>> amplitudes;
+  amplitudes.reserve(bitstrings.size());
+
+  using Runner = qsim::QuantumTrajectorySimulator<IO, Cirq::GateCirq<float>,
+                                                  MultiQubitGateFuser,
+                                                  Simulator>;
+
+  Runner::Parameter param;
+  uint64_t seed;
+  try {
+    param.num_threads = parseOptions<unsigned>(options, "t\0");
+    param.max_fused_size = parseOptions<unsigned>(options, "f\0");
+    param.verbosity = parseOptions<unsigned>(options, "v\0");
+    seed = parseOptions<unsigned>(options, "s\0");
+  } catch (const std::invalid_argument &exp) {
+    IO::errorf(exp.what());
+    return {};
+  }
+
+  StateSpace state_space(param.num_threads);
+  auto measure = [&bitstrings, &ncircuit, &amplitudes, &state_space](
+                  unsigned k, const State &state,
+                  std::vector<uint64_t>& stat) {
+    for (const auto &b : bitstrings) {
+      amplitudes.push_back(state_space.GetAmpl(state, b));
```

Comment:
Based on the name of this function `qtrajectory_simulate` why are you keep track of the raw amplitudes you get for each bitstring you draw ?

Reply:
This follows the previous naming pattern of `*_simulate` for amplitudes and `*_simulate_fullstate` for state vector. As noted in the PR description, each `qtrajectory_{foo}` method is the noisy equivalent of the `qsim_{foo}` method.